src/test/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialBindingTest.java

```diff
@@ -0,0 +1,109 @@
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    private Server server;
+    private String serverUri;
+
+    @Before
+    public void prepareFakeOAuthServer() throws Exception {
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        context.addServlet(new ServletHolder(new MockHttpServlet()), "/*");
+
+        server = new Server(0);
+        server.setHandler(context);
+        server.start();
+
+        serverUri = server.getURI().toString();
+    }
+
+    @After
+    public void unprepareFakeOAuthServer() throws Exception {
+        server.stop();
+    }
+
+    @Test
+    public void basics() throws IOException {
+        final String credentialsId = "creds";
+        final String username = "bob";
+        final String password = "s$$cr3t";
+
+        OpenShiftBearerTokenCredentialImpl t = new OpenShiftBearerTokenCredentialImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+        String token = t.getToken(serverUri + "valid-response", null, false);
+
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                OpenShiftBearerTokenCredentialImpl c = new OpenShiftBearerTokenCredentialImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), c);
+
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  withCredentials([openshiftBearerToken(oauthServerURL: '" + serverUri + "valid-response', variable: 'TOKEN', credentialsId: '" + credentialsId + "')]) {\n"
+                        + "    semaphore 'basics'\n"
+                        + "    sh 'echo $TOKEN'\n"
+                        + "  }\n"
+                        + "}", true));
+
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("basics/1", b);
+            }
+        });
+
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                assertNotNull(p);
+
+                WorkflowRun b = p.getBuildByNumber(1);
+                assertNotNull(b);
+                assertEquals(Collections.<String>emptySet(), grep(b.getRootDir(), password));
+
+                SemaphoreStep.success("basics/1", null);
+                story.j.waitForCompletion(b);
+                story.j.assertBuildStatusSuccess(b);
+                story.j.assertLogNotContains(password, b);
+
+                FilePath script = story.j.jenkins.getWorkspaceFor(p).child("script");
```

Comment:
Again, I don't know about `RestartableJenkinsRule` but I'll try to help a bit.
What should write this `script` file ? If it Jenkins that stores some local state when interrupting pipeline for a restart, it doesn't sound like something we should test as part of this plugin.

If your goal is to test if the pipeline behaves properly when being resumed from the `semaphore 'basics'` within `withCredentials`, wouldn't this tests be enough:
1. Execute the pipeline and interrupt it within `withCredentials` when to token is already available
2. Resume the pipeline and execute a script that actually stores the token into a file
3. In a third step, check that the file contains the expected token.

This definitely goes green when running the tests but maybe that's not what you want to achieve